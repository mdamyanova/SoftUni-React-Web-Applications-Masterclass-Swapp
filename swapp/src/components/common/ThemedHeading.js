import React from 'react';
import PropTypes from 'prop-types';
import { Heading } from 'rebass';

const ThemedHeading = ({ text, style }) => {
  return (
    <Heading
      sx={style}
      fontWeight={900}
      css={`
        color: var(--primary-heading-font-color);
      `}
    >
      {text}
    </Heading>
  );
};

ThemedHeading.propTypes = {
  text: PropTypes.string.isRequired,
  style: PropTypes.object
};

ThemedHeading.defaultProps = {
  style: {}
};

export default ThemedHeading;
