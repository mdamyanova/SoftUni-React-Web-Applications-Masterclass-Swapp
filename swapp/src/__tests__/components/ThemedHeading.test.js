import React from "react";
import { shallow, mount } from "enzyme";
import ThemedHeading from "../../components/common/forms/ThemedHeading";

describe("<ThemedHeading />", () => {
  it("should match snapshot", () => {
    expect(shallow(<ThemedHeading text="test" />)).toMatchSnapshot();
  });

  it("should have props", () => {
    const wrapper = mount(<ThemedHeading text="test" />);
    expect(wrapper.props().text).toEqual("test");
  });
});
