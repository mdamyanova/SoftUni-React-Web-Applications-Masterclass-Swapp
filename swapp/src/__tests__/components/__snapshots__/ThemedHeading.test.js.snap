// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ThemedHeading /> should match snapshot 1`] = `
<ForwardRef
  css="
        color: var(--primary-heading-font-color);
      "
  fontWeight={900}
  sx={Object {}}
>
  test
</ForwardRef>
`;
